src/core/mac/mac.cpp

```diff
@@ -1093,25 +1043,37 @@ void Mac::SentFrame(otError aError)
                      sendFrame.ToInfoString(stringBuffer, sizeof(stringBuffer)));
         otDumpDebgMac(GetInstance(), "TX ERR", sendFrame.GetHeader(), 16);
 
-        if (!RadioSupportsRetries() &&
-            mTransmitAttempts < sendFrame.GetMaxTxAttempts())
+        if (!RadioSupportsRetries() && mTransmitAttempts < sendFrame.GetMaxTxAttempts())
         {
-            StartCsmaBackoff();
             mCounters.mTxRetry++;
+            StartCsmaBackoff();
             ExitNow();
         }
 
         OT_UNUSED_VARIABLE(stringBuffer);
-        break;
     }
 
-    default:
-        assert(false);
-        break;
+    // Get the destination address and find the corresponding neighbor (if any).
+
+    sendFrame.GetDstAddr(dstAddr);
+    neighbor = GetNetif().GetMle().GetNeighbor(dstAddr);
+
+    // Process the ack frame for "frame pending".
+
+    if (aError == OT_ERROR_NONE && sendFrame.GetAckRequest() && aAckFrame != NULL)
+    {
+        Frame &ackFrame = *static_cast<Frame *>(aAckFrame);
+        framePending = ackFrame.GetFramePending();
+
+        if (neighbor != NULL)
+        {
+            neighbor->GetLinkInfo().AddRss(GetNoiseFloor(), ackFrame.GetRssi());
+        }
     }
 
     mTransmitAttempts = 0;
-    mCsmaAttempts = 0;
+
```

Comment:
nit: move `mTransmitAttempts = 0` to before line 1056 above?

Reply:
This was also intentional. I moved it down so we can have number of tx attempts it took to send this frame after we find the `neighbor` (again for later changes to keep track of average number of retx per frame per neighbor). 

I can move it back and change this along with the next PRs.